udata/migrations/2025-01-05-dataservices-fields-changes.py

```diff
@@ -0,0 +1,61 @@
+"""
+This migration keeps only the "Local authority" badge if the organization also has the "Public service" badge.
+"""
+
+import logging
+from typing import List
+
+from udata.core.dataservices.constants import (
+    DATASERVICE_ACCESS_TYPE_OPEN,
+    DATASERVICE_ACCESS_TYPE_OPEN_WITH_ACCOUNT,
+    DATASERVICE_ACCESS_TYPE_RESTRICTED,
+)
+from udata.core.dataservices.models import Dataservice
+
+log = logging.getLogger(__name__)
+
+
+def migrate(db):
+    log.info("Processing dataservices…")
+
+    Dataservice.objects(is_restricted=True, has_token=True).update(
```

Comment:
I don't think we cover all cases since we can have is_restricted and has_token to None.
We should make sure that `Dataservice.objects(access_type__exists=True).count() == Dataservice.objects().count()` right?

Reply:
Hum… We can have `None` in these fields?

Reply:
I did a `None` request here 6e7c39975